core/src/main/java/hudson/FileSystemProvisioner.java

```diff
@@ -23,230 +23,41 @@
- * moment for the user to specify the retention policy.
- *
- * <li>
- * Configuration mechanism. Should we auto-detect FileSystemProvisioner
- * per OS? (but for example, zfs support would require the root access.)
- * People probably needs to be able to disable this feature, which means
- * one more configuration option. It's especially tricky because
- * during the configuration we don't know the OS type.
- *
- * OTOH special agent type like the ones for network.com grid can
- * hide this.
- * </ol>
- *
- *
- * <h2>Recap</h2>
- *
- * To recap,
- *
- * - when an agent connects, we auto-detect the file system provisioner.
- *   (for example, ZFS FSP would check the agent root user prop
- *   and/or attempt to "pfexec zfs create" and take over.)
- *
- * - the user may configure jobs for snapshot collection, along with
- *   the retention policy.
- *
- * - keep workspace snapshots that correspond to the permalinks
- *   In ZFS, use a user property to remember the build and the job.
- *
- * Can't the 2nd step happen automatically, when someone else depends on
- * the workspace snapshot of the upstream? Yes, by using {@link RunListener}.
- * So this becomes like a special SCM type.
- *
- *
- *
- * <h2>Design take 2</h2>
- * <p>
- * The first piece of this is the custom {@link SCM}, which inherits the
- * workspace of another job. When this executes, it picks up
- * {@link WorkspaceSnapshot} from the other job and use it to obtain the workspace.
- *
- * <p>
- * Then there's {@link RunListener}, which creates a snapshot if
- * someone else is interested in using a snapshot later.
- *
- * <h3>TODOs</h3>
- * <ul>
- * <li>
- * Garbage collection of workspace snapshots. 
- *
- * </ul>
- *
- * @author Kohsuke Kawaguchi
- * @since 1.235
+ * @deprecated Unused.
  */
-public abstract class FileSystemProvisioner implements ExtensionPoint, Describable<FileSystemProvisioner> {
-    /**
-     * Called very early in the build (before a build places any files
-     * in the workspace, such as SCM checkout) to provision a workspace
-     * for the build.
-     *
-     * <p>
-     * This method can prepare the underlying file system in preparation
-     * for the later {@link FileSystemProvisioner.Default#snapshot(AbstractBuild, FilePath, TaskListener)}.
-     *
-     * TODO : the method needs to be able to see the snapshot would
-     * be later needed. In fact, perhaps we should only call this method
-     * when Hudson knows that a snapshot is later needed?
-     *
-     * @param ws
-     *      New workspace should be prepared in this location. This is the same value as
-     *      {@code build.getProject().getWorkspace()} but passed separately for convenience.
-     */
+@Deprecated
+public abstract class FileSystemProvisioner implements Describable<FileSystemProvisioner> {
     public abstract void prepareWorkspace(AbstractBuild<?,?> build, FilePath ws, TaskListener listener) throws IOException, InterruptedException;
```

Comment:
Have to keep the class and this method since it is called by: https://github.com/jenkinsci/compound-slaves-plugin/blob/e8e8aef6fe97cf5528883343928e11dc9adbfd39/src/main/java/ru/yandex/jenkins/plugins/compound/CompoundBuilder.java#L71